### DISCUSSION TITLE
Sha256sum of Repository-master.zip

### DISCUSSION THREAD

**Discussion by Vladmaster99** [community/community#30467] [Actions]
I am sure this has been asked before but how do you get the Sha256 of a downloaded source.zip file. the sha256/Sha1?  value is made up of the blobs of the files on github not the sha256 of the downloaded source.zip. I have seen this question asked before on other websites but never seen how to get the sha256 value off github. This is very easy to do on Sourceforge. which is why I alway check sourceforge first.

I have search the github community pages looking up sha1 and sha256 I did not find the answer only that git.exe had now got sha256 support on it new list of features. but that github still uses sha1 I think

**Comment by airtower-luna** [comment]
I think you're confusing ZIP checksums with Git commit IDs. GitHub doesn't provide checksums  for ZIP downloads.

Git objects IDs (including commit IDs) are usually SHA-1 hashes, and there is *experimental* code for SHA-256 object IDs. These hashes provide an integrity check for the repository, but you need to clone the repository instead of just downloading a ZIP to take advantage of that. For how Git object IDs work, see: [Git Internals - Git Objects](https://git-scm.com/book/en/v2/Git-Internals-Git-Objects)

**Comment by Vladmaster99** [comment]
Yes you understand-ed my question exactly ... I am looking to get the zip "AND" the  GZip  checksums directly from github ( aka like souceforge) . I understand that github has at the moment SHA-1 hashes for its internal git use and will sometimes in the future move to Sha-1-256 hashes with a hybrid approach of using a table of both Sha-1 and sha-1-256 . But it does yet have a simple feature (like sourceforge) of supplying md5,Sha1,Sha256 or any other hashes for its zip and gziped files for users that are looking to use code and are not looking at writing or patching the source ..but just using the program,.library,source  etc for their own project/program. Thanks for your response. It may clarify the question to others but does not answer the question of if or when git hub will provide zip and Gzipped hashes? ..Thanks for responding airtower-luna and have a wonderful week. cheers

**Reply by Vladmaster99** [reply]
But it does yet have -->But it does **not** yet have .. sorry for the errror . Everybody have a great year ...Cheers

**Reply by Vladmaster99** [reply]
grammer nazi here understand-ed should be UNDERSTOOD. opps double crap. me bad... {yes correct me if you must ;-) }
